Changing CMake files after configuring leads to build failure

Greetings, when doing something like:
```
❯ cmake -B build -S . -DBRION_REQUIRE_PYTHON=ON
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Boost: /usr/include (found version "1.76.0") found components: date_time filesystem iostreams program_options regex system unit_test_framework 
…[snip]…
-- Skipping Brion Tests. Test generation requires test data
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/Brion/build
```
to get a successfully configured build directory, and then updating some of the CMake:
```
❯ touch CMakeLists.txt 
```
I would then expect a build to trigger a CMake re-execution that should run through with cached values and resume the build. Instead, the CMake re-execution fails:
```
❯ cmake --build build -j4
-- Found Boost: /usr/include (found version "1.76.0") found components: date_time filesystem iostreams program_options regex system unit_test_framework 
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Could NOT find Boost (missing: python) (found version "1.76.0")
-- Version: 7.1.2
-- Build type: 
-- CXX_STANDARD: 11
-- Required features: cxx_variadic_templates
CMake Error at brain/python/CMakeLists.txt:14 (message):
  Not configured brion.py: Boost.Python, Python or Numpy not found.  Build
  set to force python


-- Configuring incomplete, errors occurred!
See also "/tmp/Brion/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/Brion/build/CMakeFiles/CMakeError.log".
gmake: *** [Makefile:898: cmake_check_build_system] Error 1
```
leading me to have to remove the build directory and starting from scratch.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,9 +69,6 @@ else()
     set(BRION_USE_CUSTOM_LIB_DIR ON)
 endif()
 
-# Configure used python version
-set(USE_PYTHON_VERSION 3)
-
 # Python wheels needed variables
 set(BRION_DESCRIPTION "The Blue Brain C++ I/O library")
 set(BRION_MAINTAINER_NAME "Blue Brain Project")

